frontend/catalyst/api_extensions/control_flow.py

```diff
@@ -237,8 +238,17 @@ def conditional_fn():
     """
 
     def _decorator(true_fn: Callable):
+
         if len(inspect.signature(true_fn).parameters):
-            raise TypeError("Conditional 'True' function is not allowed to have any arguments")
+            if isinstance(true_fn, type) and issubclass(true_fn, qml.operation.Operation):
```

Comment:
Is the second part not sufficient?

Reply:
No, because the callable can be an instance instead of a class.